ape_foundry/provider.py

```diff
@@ -991,3 +979,31 @@ def reset_fork(self, block_number: Optional[int] = None):
         # # Rest the fork
         result = self._make_request("anvil_reset", [{"forking": forking_params}])
         return result
+
+
+# Abstracted for easier testing conditions.
+def _extract_custom_error(provider: FoundryProvider, **kwargs) -> str:
+    data = {}
+    if "trace" in kwargs:
+        kwargs["trace"], new_trace = tee(kwargs["trace"])
+        data = list(new_trace)[-1].raw
+
+    elif "txn" in kwargs:
+        txn = kwargs["txn"]
+        txn_hash = txn.txn_hash if isinstance(txn.txn_hash, str) else txn.txn_hash.hex()
+        try:
+            # NOTE: Disable memory because large input data fails
+            #    And all we need is the revert message.
```

Comment:
question: Disable memory?  Did you mean to have `enable_memory=False` in the call below or something?

Reply:
I was going to do a flow like that - because I noticed problems with memory and large calldata.. (in addition to the hex-str bug), but I realized it made no sense because you need memory to get the return values anyway, so I removed that.

#fb4bcfc cleans it up